src/settings/settings.ts

```diff
@@ -44,6 +44,21 @@ export const appearanceSettings = createGlobalSettings<Appearance>('apperance',
     primary: () => i18n.t('settings_appearance'),
 })
 
+export enum WholePostVisibility {
+    all = 'all',
+    enhancedOnly = 'enhancedOnly',
```

Comment:
enhanceableOnly

Reply:
+ `enhancedOnly` - For a post that was enhanced by Maskbook plugin.
+ `encryptedOnly` - For a post that was encrypted by Maskbook.

Reply:
After you replace the original content with Maskbook one, it is "enhanced". otherwise, it only have possibility to be enhanced.

Reply:
The post already enhanced/encrypted before presenting it to the end-user. The `WholePostVisibility` setting determines whether or not to use `<WhilePostComponent/>` to display the post.

Reply:
@Jack-Works I deleted your comment by mistake.

Reply:
this explanation makes sense to me.